### PR TITLE
add tests for prepareForDataSynchronization func

### PR DESCRIPTION
**Pull request by peng225**
Signed-off-by: Shinya Hayashi <shinya-hayashi@cybozu.co.jp>

### CODE REVIEW

**Review by satoru-takeuchi — CHANGES_REQUESTED** [review]

File: internal/controller/mantlebackup_controller_test.go [hunk 1]
```diff
@@ -534,16 +485,287 @@ var _ = Describe("searchDiffOriginMantleBackup", func() {
...
+			}).Times(1).Return(
+			&proto.ListMantleBackupResponse{
+				MantleBackupList: data,
+			}, nil)
+
+		ctrlClient := fake.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+		Expect(ctrlClient).NotTo(BeNil())
+		primaryBackups := append(primaryBackupsWithoutTarget, backup.DeepCopy())
+		for _, backup := range primaryBackups {
+			shouldbeDeleted := false
```
[the remark this answers is not in the document]
Comment (peng225) [reply internal/controller/mantlebackup_controller_test.go]: Fixed.
https://github.com/cybozu-go/mantle/compare/422e9f9fd8c354bf96b67a9b80d8789737acf645..a0e5141623f5e8a82fbe426f28d81cba85aa0ed1

File: internal/controller/mantlebackup_controller_test.go [hunk 2]
```diff
@@ -547,3 +550,274 @@ var _ = Describe("searchDiffOriginMantleBackup", func() {
...
+	meta.SetStatusCondition(&newMB.Status.Conditions, metav1.Condition{
+		Type:   mantlev1.BackupConditionSyncedToRemote,
+		Status: syncedToRemote,
+	})
+	return newMB
+}
+
+var _ = Describe("prepareForDataSynchronization", func() {
+	testPVCUID := "d3b07384-d9a7-4e6b-8a3b-1f4b7b7b7b7b"
+	testName := "test5"
```
[the remark this answers is not in the document]
Comment (peng225) [reply internal/controller/mantlebackup_controller_test.go]: Fixed.
https://github.com/cybozu-go/mantle/compare/422e9f9fd8c354bf96b67a9b80d8789737acf645..a0e5141623f5e8a82fbe426f28d81cba85aa0ed1

**Comment by satoru-takeuchi** [comment]
Although `build` workflow failed, it seems not to be relatated to this PR. The test fails on creating minikube cluster. I re-ran this workflow.

**Review by satoru-takeuchi — APPROVED** [review]
@peng225 LGTM. Do you plan to squash commits? If else, I'll merge this PR after CI succeeded.

**Comment by peng225** [comment]
> Do you plan to squash commits?

No, I don't. Now that CI has passed, I'll merge it.

**Pull request merged**